tests/sync/test_synchronizer.py

```diff
@@ -66,9 +66,50 @@ def run(x, y):
         'killed': False,
         'defaultTreatment': 'off',
         'algo': 2,
-        'conditions': []
+        'conditions': [{
+            'conditionType': 'WHITELIST',
+            'matcherGroup':{
+                'combiner': 'AND',
+                'matchers':[{
+                    'matcherType': 'IN_SEGMENT',
+                    'negate': False,
+                    'userDefinedSegmentMatcherData': {
+                        'segmentName': 'segmentA'
+                    }
+                }]
+            },
+            'partitions': [{
+                'size': 100,
+                'treatment': 'on'
+            }]
+        }]
     }]
 
+    def test_synchronize_splits(self, mocker):
```

Comment:
i think it might be worth adding some tests for 
- `_fetch_until` checking that the returned value is correct
- ensuring that `segment_sync` gets called once for every newly referenced segment

Reply:
First suggestion is applied in the test_synchronize_splits function, as the segment name is picked up from the split definitions.
Will create the second test.